packages/custom-calendar-builder/src/calendar-builder-app.ts

```diff
@@ -0,0 +1,411 @@
+          } catch (error) {
+            console.error('Failed to load calendar file:', error);
+            ui.notifications?.error(game.i18n.localize('CALENDAR_BUILDER.app.notifications.import_failed'));
+          }
+        },
+      });
+
+      await filePicker.render(true);
+    } catch (error) {
+      console.error('Failed to open file picker:', error);
+      ui.notifications?.error('Failed to open file picker');
+    }
+  }
+
+  /**
+   * Save calendar to file action
+   */
+  async _onSaveCalendar(event: Event, target: HTMLElement): Promise<void> {
+    if (!this.currentJson.trim()) {
+      ui.notifications?.warn(game.i18n.localize('CALENDAR_BUILDER.app.notifications.no_content'));
+      return;
+    }
+
+    // For now, just export as JSON
+    this._onExportJson(event, target);
+  }
+
+  /**
+   * Export JSON to file action
+   */
+  async _onExportJson(event: Event, target: HTMLElement): Promise<void> {
+    if (!this.currentJson.trim()) {
+      ui.notifications?.warn(game.i18n.localize('CALENDAR_BUILDER.app.notifications.no_content'));
+      return;
+    }
+
+    try {
+      // Create blob and download
+      const blob = new Blob([this.currentJson], { type: 'application/json' });
+      const url = URL.createObjectURL(blob);
+
+      const a = document.createElement('a');
+      a.href = url;
+      a.download = 'custom-calendar.json';
+      document.body.appendChild(a);
+      a.click();
+      document.body.removeChild(a);
+      URL.revokeObjectURL(url);
+
+      ui.notifications?.info(game.i18n.localize('CALENDAR_BUILDER.app.notifications.exported'));
+    } catch (error) {
+      console.error('Export failed:', error);
+      ui.notifications?.error(game.i18n.localize('CALENDAR_BUILDER.app.notifications.export_failed'));
+    }
+  }
+
+  /**
+   * Import JSON from clipboard or file input
+   */
+  async _onImportJson(event: Event, target: HTMLElement): Promise<void> {
+    // Create a temporary file input
+    const input = document.createElement('input');
+    input.type = 'file';
+    input.accept = '.json';
+
+    input.addEventListener('change', async (e) => {
+      const file = (e.target as HTMLInputElement).files?.[0];
+      if (!file) return;
+
+      try {
+        const text = await file.text();
+        // Validate it's proper JSON
+        JSON.parse(text);
+
+        this.currentJson = text;
+        this.render(true);
+        ui.notifications?.info(game.i18n.localize('CALENDAR_BUILDER.app.notifications.imported'));
+      } catch (error) {
+        console.error('Import failed:', error);
+        ui.notifications?.error(game.i18n.localize('CALENDAR_BUILDER.app.notifications.invalid_json'));
+      }
+    });
+
+    input.click();
+  }
+
+  /**
+   * Validate JSON action
+   */
+  async _onValidateJson(event: Event, target: HTMLElement): Promise<void> {
+    await this._validateCurrentJson();
+    ui.notifications?.info(game.i18n.localize('CALENDAR_BUILDER.app.notifications.validation_complete'));
```

Comment:
All notifications for this should be done through the application window, not `ui.notifications`